common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/DownloadFile.java

```diff
@@ -27,12 +27,10 @@
  */
 public interface DownloadFile {
   /**
-   * Delete the file.
-   *
-   * @return  <code>true</code> if and only if the file or directory is
-   *          successfully deleted; <code>false</code> otherwise
+   * Creates a new {@link DownloadFileDeleter} for the underlying resources of
+   * the {@link DownloadFile}.
    */
-  boolean delete();
+  DownloadFileDeleter deleter();
```

Comment:
Can we just make this as a `Callable` variable without introducing `DownloadFileDeleter`?

Reply:
Yes, that's a good alternative, but in this case we'd have to deal with `throws Exception` in its signature.
The client side would look like this then:
```
  try {
    targetFile.deleter().call();
  } catch (Exception e) {
    throw new IOException("Failed to delete file", e);
  }
```
Is there a different `Callable` interface without checked `Exception` declaration?

Reply:
There is simply `Function`?

Reply:
Missed the comment and now it's already outdated, but just for the record...
I tried to use `Function` first, but it turned out to be ugly due to the mix of Java and Scala.

Reply:
Oh I meant Java's `Function` but no big deal anyway

Reply:
Right, that's what I tried, but one of the implementations of the Java interface is in Scala (`EncryptedDownloadFile`).
So it looked something like:
```
def deleter(): java.util.function.Function[java.lang.Void, java.lang.Boolean] = ...
```
and on the call side:
```
file.deleter().apply(null)
```
It feels like there has to be an easy solution for the interoperability problem, but nothing comes to mind except to have mirroring Scala interfaces or implicit, like
```
implicit def convertFunc(
      f: java.util.function.Function[java.lang.Void, java.lang.Boolean]): () => Boolean =
    () => f.apply(null)
```

BTW, thanks for reviewing!